diff --git a/arch/arm64/boot/dts/xilinx/zynqmp.dtsi b/arch/arm64/boot/dts/xilinx/zynqmp.dtsi
index e7d305f..c240858 100644
--- a/arch/arm64/boot/dts/xilinx/zynqmp.dtsi
+++ b/arch/arm64/boot/dts/xilinx/zynqmp.dtsi
@@ -420,10 +420,11 @@
 	};
 
 	smmu: smmu@fd800000 {
-		compatible = "arm,mmu-500";
+		//compatible = "arm,mmu-500";
+		compatible = "i3,smmu-1";
 		reg = <0x0 0xfd800000 0x0 0x20000>;
 		#iommu-cells = <1>;
-		status = "disabled";
+		status = "okay";
 		#global-interrupts = <1>;
 		interrupt-parent = <&gic>;
 		interrupts = <0 155 4>,
@@ -724,7 +725,7 @@
 			#address-cells = <1>;
 			#size-cells = <0>;
 			#stream-id-cells = <1>;
-			iommus = <&smmu 0x872>;
+			/* iommus = <&smmu 0x872>; */
 			power-domains = <&pd_nand>;
 		};
 
@@ -738,7 +739,7 @@
 			#address-cells = <1>;
 			#size-cells = <0>;
 			#stream-id-cells = <1>;
-			iommus = <&smmu 0x874>;
+			/* iommus = <&smmu 0x874>; */
 			power-domains = <&pd_eth0>;
 		};
 
@@ -752,7 +753,7 @@
 			#address-cells = <1>;
 			#size-cells = <0>;
 			#stream-id-cells = <1>;
-			iommus = <&smmu 0x875>;
+			/* iommus = <&smmu 0x875>; */
 			power-domains = <&pd_eth1>;
 		};
 
@@ -766,7 +767,7 @@
 			#address-cells = <1>;
 			#size-cells = <0>;
 			#stream-id-cells = <1>;
-			iommus = <&smmu 0x876>;
+			/* iommus = <&smmu 0x876>; */
 			power-domains = <&pd_eth2>;
 		};
 
@@ -780,7 +781,7 @@
 			#address-cells = <1>;
 			#size-cells = <0>;
 			#stream-id-cells = <1>;
-			iommus = <&smmu 0x877>;
+			/* iommus = <&smmu 0x877>; */
 			power-domains = <&pd_eth3>;
 		};
 
@@ -948,7 +949,7 @@
 			#address-cells = <1>;
 			#size-cells = <0>;
 			#stream-id-cells = <1>;
-			iommus = <&smmu 0x873>;
+			/* iommus = <&smmu 0x873>; */
 			power-domains = <&pd_qspi>;
 		};
 
@@ -1015,7 +1016,7 @@
 			clock-names = "clk_xin", "clk_ahb";
 			xlnx,device_id = <0>;
 			#stream-id-cells = <1>;
-			iommus = <&smmu 0x870>;
+			/* iommus = <&smmu 0x870>; */
 			power-domains = <&pd_sd0>;
 			nvmem-cells = <&soc_revision>;
 			nvmem-cell-names = "soc_revision";
@@ -1031,7 +1032,7 @@
 			clock-names = "clk_xin", "clk_ahb";
 			xlnx,device_id = <1>;
 			#stream-id-cells = <1>;
-			iommus = <&smmu 0x871>;
+			/* iommus = <&smmu 0x871>; */
 			power-domains = <&pd_sd1>;
 			nvmem-cells = <&soc_revision>;
 			nvmem-cell-names = "soc_revision";
@@ -1142,7 +1143,7 @@
 				interrupt-parent = <&gic>;
 				interrupts = <0 65 4>, <0 69 4>, <0 75 4>;
 				#stream-id-cells = <1>;
-				iommus = <&smmu 0x860>;
+				/* iommus = <&smmu 0x860>; */
 				snps,quirk-frame-length-adjustment = <0x20>;
 				snps,refclk_fladj;
 				snps,enable_guctl1_resume_quirk;
@@ -1172,7 +1173,7 @@
 				interrupt-parent = <&gic>;
 				interrupts = <0 70 4>, <0 74 4>, <0 76 4>;
 				#stream-id-cells = <1>;
-				iommus = <&smmu 0x861>;
+				/* iommus = <&smmu 0x861>; */
 				snps,quirk-frame-length-adjustment = <0x20>;
 				snps,refclk_fladj;
 				snps,enable_guctl1_resume_quirk;
