diff --git a/drivers/iommu/Kconfig b/drivers/iommu/Kconfig
index c60395b..356fbed 100644
--- a/drivers/iommu/Kconfig
+++ b/drivers/iommu/Kconfig
@@ -344,6 +344,20 @@ config ARM_SMMU
 	  Say Y here if your SoC includes an IOMMU device implementing
 	  the ARM SMMU architecture.
 
+# ARM IOMMU support with SVM
+config ARM_SMMU_SVM
+	bool "ARM Ltd. System MMU (SMMU) Support with SVM"
+	depends on ARM64 && MMU
+	select IOMMU_API
+	select IOMMU_IO_PGTABLE_LPAE
+	select MMU_NOTIFIER
+	help
+	  SVM support for implementations of the ARM System MMU architecture
+	  versions 1 and 2.
+
+	  Say Y here if your SoC includes an IOMMU device implementing
+	  the ARM SMMU architecture and you wish to have SVM support.
+
 config ARM_SMMU_V3
 	bool "ARM Ltd. System MMU Version 3 (SMMUv3) Support"
 	depends on ARM64
